ssl/record/ssl3_record.c

```diff
@@ -663,6 +663,15 @@ int ssl3_get_record(SSL *s)
             goto f_err;
         }
 
+        /* If client see a packet overflowing his requested MFL setting */
+        if (!s->server && s->session != NULL
+                && USE_MAX_FRAGMENT_LENGTH_EXT(s->session)
+                && thisrr->length > GET_MAX_FRAGMENT_LENGTH(s->session)) {
```

Comment:
Bad assumption from my debug sessions ...
Without it, I get a :
`# 5136:error:14068044:SSL routines:do_ssl3_write:internal error:ssl\record\rec_layer_s3.c:830:`

Last test of previous If is falling when max_fragment_size is lower than active MFL limit :
```
...
                    && !WPACKET_reserve_bytes(thispkt, maxcomplen,
                                              &compressdata))) {
            SSLerr(SSL_F_DO_SSL3_WRITE, ERR_R_INTERNAL_ERROR);

```